a1/src/main/java/ro/utcn/sd/it/a1/Controller/ConsoleController.java

```diff
@@ -0,0 +1,131 @@
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Optional;
+import java.util.Scanner;
+@Component
+@RequiredArgsConstructor
+public class ConsoleController implements CommandLineRunner {
+    private final Scanner scanner=new Scanner(System.in);
+    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+   private User user = null;
+   private Tag tag=null;
+
+    private final QuestionManagementSystem qms;
+
+
+    @Override
+    public void run(String... args)  {
+        print("Welcome!");
+        boolean done=false;
+        while(!done){
+            print("Enter a command: ");
+            String command=scanner.next().trim();
+            try{
+                done=handleCommand(command);
+            }catch (QuestionNotFoundException | IOException questionNotFoundEx){
+                print("Error! ");
+            }
+
+        }
+
+
+    }
+    private boolean handleCommand(String command) throws IOException {
+        switch(command){
+            case "login":
+                loginuser();
+                return false;
+            case "ask_question":
+                askQ();
+                return false;
+            case "display_questions":
+                displayQ();
+                return false;
+            case "filter_question":
+                filterQ();
+                return false;
+            default:
+                print("Unknown command! ");
+                return false;
+        }
+    }
+
+    private void displayQ(){
+
+        qms.listQuestionByDate().forEach(s->print(s.toString()));
+    }
+
+    private void loginuser(){
+
+        print("Enter username: ");
+        String username=scanner.next().trim();
+
+        print("Enter password: ");
+        String password=scanner.next().trim();
+        Optional<User> user =qms.login(username,password) ;
+        if(user.isPresent()) {
+            this.user=user.get();
+            print("YOu are logged in! ");
+        }else
+            print("Wrong username or password");
+
+
+
+    }
+    private void askQ() throws IOException {
+        if(user == null){
+           print("You need to be logged in!");
+        }else {
+            print("Title: ");
+            String title = br.readLine();
+            print("Text: ");
+            String text = br.readLine();
+            //print("Author: ");
+            //  String author =scanner.next().trim();
+             print("Tag: ");
+             String t=scanner.next().trim();
+            // print("Date: ");
+            Timestamp date_time = new Timestamp(System.currentTimeMillis());
+            // String date = new SimpleDateFormat("dd-MM-yyyy").format(new Date())
+            // String date=scanner.next().trim();
+            //Tag tag= qms.addTag(new Tag(t));
+           // print("Your tag is :"+tag);
+            Tag tag=new Tag(t);
```

Comment:
Also, you have  some commented out code. Usually you would simply delete this code (but commit more often) and have a history of your code in git.